buildpack/centos8/Dockerfile

```
@@ -51,10 +51,9 @@ RUN yum -y install \
 
 
 # Python and tools
-RUN yum -y install python3 python3-devel python3-wheel && \
-    wget -qO - https://bootstrap.pypa.io/get-pip.py | python3 && \
-    pip install --upgrade "pip==19.1.1" && \
-    pip install setuptools virtualenv && \
+RUN yum -y install python3 python3-devel python3-wheel python3-virtualenv && \
+    pip3 install --user --upgrade pip==20.0.2 && \
```

```suggestion
    pip3 install --user --upgrade pip==19.1.1 && \
```
to keep the same version across all builds.

---

And yes, indeed with the `--user` flag it works well.

After research it looks like related to https://github.com/pypa/pip/issues/5373 when the default package-distributed pip does something strange that prevents upgrading pip via pip.